mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -491,8 +492,7 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
         if (length == 0) {
             LOGGER.warn("Ignoring mailbox with empty name");
         } else {
-            if (mailboxPath.getName().charAt(length - 1) == getDelimiter())
-                mailboxPath.setName(mailboxPath.getName().substring(0, length - 1));
+            mailboxPath.setName(getDelimiter().removeTrailingSeparatorAtTheEnd(mailboxPath.getName()));
```

Comment:
it's MailboxPath responsibility IMO

Reply:
See mailboxPath.sanitize(char pathDelimiter) as handled in https://github.com/linagora/james-project/pull/1109 ...

Maybe doing the work twice is not worth it?